app_clients/cassandra-client/cassandra-liveness-check.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+
+#####################
+#  VAR DEFINITION   #
+#####################
+
+: << EOF
+The below variables are derived from env, with 
+default values specified where the env are not present
+EOF
+
+KEYSPACE_PREFIX="Inventory"
+KEYSPACE_SUFFIX=`echo $(mktemp) | cut -d '.' -f 2`
+KEYSPACE_NAME="${KEYSPACE_PREFIX}_${KEYSPACE_SUFFIX}"
+TABLE_PREFIX="Dataset"
+TABLE_SUFFIX=`echo $(mktemp) | cut -d '.' -f 2`
+TABLE_NAME="${TABLE_PREFIX}_${TABLE_SUFFIX}"
+
+# Cassandra Port
+Port=${CASSANDRA_PORT:-9042}
+
+# Liveness probe failure timeout 
+Svc=${CASSANDRA_SVC_NAME:-cassandra}
+
+# Liveness check interval
+L_P_S=${LIVENESS_PERIOD_SECONDS:-10}
+
+# Liveness probe failure timeout 
+L_T_S=${LIVENESS_TIMEOUT_SECONDS:-10}
+
+# Liveness probe failure retry count
+L_R_C=${LIVENESS_RETRY_COUNT:-3}
+
+###################
+#   FUNCTIONS     #
+###################
+
+# Describe script usage 
+usage()
+{
+ if [[ $1 = "--help" || $1 = "-h" ]]; then
+     echo 
+     echo "Usage: bash $0"
+     echo
+     echo "Set these MANDATORY env before running the script"
+     echo 
+     echo "CASSANDRA_PORT              : Cassandra Transport Port"
+     echo "CASSANDRA_SVC_NAME          : Cassandra Service Name"
+     echo
+     echo "Set these Optional env before running the script"
+     echo
+     echo "LIVENESS_PERIOD_SECONDS : Liveness check interval (default: 10s)"
+     echo "LIVENESS_TIMEOUT_SECONDS: Liveness probe failure timeout (default: 10s)"
+     echo "LIVENESS_RETRY_COUNT    : Liveness probe failure retry count (default: 3)"
+     echo 
+     exit  
+ fi  
+}
+
+# Stores status in a action.status file
+function updateStatus(){
+    status=$1 
+    echo "status is: ${status}"
+    echo ${status} > /var/tmp/action.status
+}
+
+# Create the keyspace
+create_keyspace()
+{
+  echo "creating the keyspace"
+
+  for i in `seq 1 $L_R_C`; do
+   cqlsh $1 $2 -e "CREATE KEYSPACE IF NOT EXISTS $KEYSPACE_NAME WITH replication = {'class' : 'SimpleStrategy', 'replication_factor' : '3'};" > /dev/null 2>&1
```

Comment:
take REPLICATION_FACTOR as ENV var from experiment (i.e., as an engine override depending on the instance's deployment).